src/conv-core/cpuaffinity.C

```diff
@@ -602,7 +602,16 @@ static void cpuAffinityRecvHandler(void *msg)
     if (m->node_hosts[i] == myhost)
       ++nodes_on_host;
 
-  const int package_count = cmi_hwloc_get_nbobjs_by_type(topology, HWLOC_OBJ_PACKAGE);
+  // Filter out packages with null cpusets, caused by system resource management
+  const int total_package_count = cmi_hwloc_get_nbobjs_by_type(topology, HWLOC_OBJ_PACKAGE);
+  std::vector<hwloc_obj_t> packages;
```

Comment:
If we expect packages to usually not contain null cpusets, then it may be good to reserve 'total_package_count' space in the vector before doing the push_back's in the loop below to avoid potential reallocations

Reply:
Done